Process.c

```diff
@@ -513,12 +513,16 @@ void Process_toggleTag(Process* this) {
 }
 
 bool Process_setPriority(Process* this, int priority) {
-   int old_prio = getpriority(PRIO_PROCESS, this->pid);
-   int err = setpriority(PRIO_PROCESS, this->pid, priority);
-   if (err == 0 && old_prio != getpriority(PRIO_PROCESS, this->pid)) {
-      this->nice = priority;
+   if ( Process_getuid == 0 || Process_getuid == (int) this->st_uid ) {
```

Comment:
December 7 2015 12:03 AM, "Hisham Muhammad"  wrote:
	In Process.c (https://github.com/hishamhm/htop/pull/315#discussion_r46777780):

	> @@ -513,12 +513,16 @@ void Process_toggleTag(Process* this) { > } > > bool Process_setPriority(Process* this, int priority) { > - int old_prio = getpriority(PRIO_PROCESS, this->pid); > - int err = setpriority(PRIO_PROCESS, this->pid, priority); > - if (err == 0 && old_prio != getpriority(PRIO_PROCESS, this->pid)) { > - this->nice = priority; > + if ( Process_getuid == 0 || Process_getuid == (int) this->st_uid ) { 

	My question is whether the wrapper is needed. Without the wrapping if, htop will try to change permissions, but for non-root users it will simply fail. I'd rather keep it without the wrapping if, because the reason why setpriority fails or doesn't fail may have to do with more variables in different OSes (capabilities, etc.) and not just whether the uid is the user or root.When htop runs SUID root, setpriority does *not* fail for non-root users, so a non-root user can change process priority and send signals to arbitrary processes. I guess the if doesn't make that intention clear enough. It's probably clearer if the "dangerous" code is wrapped between seteuid() calls. What do you think?